xtask/src/build_docker_image.rs

```diff
@@ -179,7 +180,17 @@ pub fn build_docker_image(
         docker_build.args(&["buildx", "build"]);
         docker_build.current_dir(&docker_root);
 
-        docker_build.args(&["--platform", &platform.docker_platform()]);
+        // add our platform, and determine if we need to use a native docker image
+        let docker_platform = platform.docker_platform();
+        let mut dockerfile = dockerfile.clone();
+        docker_build.args(&["--platform", &docker_platform]);
+        docker_build.args(&[
+            "--build-arg",
+            &format!("CROSS_TRIPLE={}", env!("CROSS_TRIPLE")),
```

Comment:
Why is this needed?

Reply:
my concern is that `CROSS_TRIPLE` is the builders platform

Reply:
@Emilgardis It's used to set the environment variable for the runner, since we might want to allow running the binaries via Qemu:

```dockerfile
ENV CROSS_TARGETARCH=$TARGETARCH
ENV CROSS_TARGETVARIANT=$TARGETVARIANT
ENV CARGO_TARGET_${CROSS_TRIPLE}_RUNNER="/native-linux-runner"
```
and

```bash
#!/usr/bin/env bash

set -eo pipefail

# shellcheck disable=SC1091
. lib.sh

main() {
    local arch
    arch=$(docker_to_linux_arch "${CROSS_TARGETARCH}" "${CROSS_TARGETVARIANT}")
    exec /linux-runner "${arch}" "${@}"
}

main "${@}"

```

Reply:
if I run this from a aarch64 system, the `CROSS_TRIPLE` will be different, is that the intended effect?

Reply:
> my concern is that `CROSS_TRIPLE` is the builders platform

I'll try to think of a better way: this is also partially an issue with the ARM detection as well.

Reply:
> if I run this from a aarch64 system, the `CROSS_TRIPLE` will be different, is that the intended effect?

Yes but also no. The `CROSS_TRIPLE` should change based on the host architecture, but this will fail if cross-compiling `cross` or if the cross triple isn't `*-linux-gnu`. So yes, but this logic doesn't work.

Reply:
> I guess we could always use `&target.nam`

I think that or `platform.target` makes sense, they should both be the same.

Reply:
I think `platform.target` is the right solution, otherwise we duplicate the work done in detecting the target platform, getting the default target, etc.

Reply:
Actually, `platform.target` doesn't work, because we actually want the target provided, so it should actually just be `target.name`, as simple as that is.

Reply:
Ok this has been solved: we should have been using `target.triple()` anyway, since we want to check if the target platform matches the Docker image platform.